docs/dev/stream/operators/process_function.md

```diff
@@ -271,9 +271,9 @@ override def onTimer(timestamp: Long, ctx: OnTimerContext, out: Collector[OUT]):
 </div>
 </div>
 
-## Optimizations
+## Timers
 
-### Timer Coalescing
+### Optimizations - Timer Coalescing
 
 Every timer registered at the `TimerService` via `registerEventTimeTimer()` or
```

Comment:
Move the first paragraph under the `## Timer`` section

Reply:
Also it would be great if you could find a good spot to add a note that calls to `processElement()` and `onTimer()` are always synchronized, i.e., users do not have to worry about concurrent modification of state.